### PR TITLE
Remove old implementation of command_argument_count

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: #4593

### CODE REVIEW

**Review by Pranavchiku — APPROVED** [review]

File: src/libasr/asr_utils.cpp [hunk 1]
```diff
@@ -1688,6 +1688,17 @@ int64_t compute_trailing_zeros(int64_t number, int64_t kind) {
...
+int32_t _argc;
+char** _argv;
+
+int32_t get_command_argument_count() {
+    int32_t result = 0;
+    for(int i=1; i<=_argc; i++) {
+        result++;
+    }
+    return result;
+}
```
Comment (certik) [inline src/libasr/asr_utils.cpp:1700]: This is incorrect, right? `command_argument_count` is a runtime function to return how many arguments were passed at runtime.

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -1194,48 +1194,28 @@ namespace CommandArgumentCount {
...
             x.base.base.loc, diagnostics);
     }
 
-    static inline ASR::asr_t* create_CommandArgumentCount(Allocator& al, const Location& loc, Vec<ASR::expr_t*>& /*args*/, diag::Diagnostics& diag) {
-        ASR::ttype_t *return_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4));
+    static ASR::expr_t *eval_CommandArgumentCount(Allocator &al, const Location &loc,
+            ASR::ttype_t *t1, Vec<ASR::expr_t*> &/*args*/, diag::Diagnostics& /*diag*/) {
+        ASRBuilder b(al, loc);
+        int64_t cnt = ASRUtils::get_command_argument_count();
+        return make_ConstantWithType(make_IntegerConstant_t, cnt, t1, loc);
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:1205]: This function doesn't have compile time implementation, only runtime.

**Comment by certik** [comment]
I think this is implemented incorrectly, so we should either revert it, or fix it.

**Comment by HarshitaKalani** [comment]
Sorry for this. I'll just revert it back.

**Comment by HarshitaKalani** [comment]
Btw, how do we implement intrinsics which support just the runtime arguments? Since get_argument_count() takes 0 input arguments, we treat it as having compile time args I think and hence, it does not go inside the instantiate_commandArgumentCount function.

**Comment by certik** [comment]
It's an intrinsic that only has runtime implementation. So we need to skip the compile time logic for it, only do the instantiation.

**Pull request merged**